src/dials/util/export_shelx.py

```diff
@@ -169,161 +170,36 @@ def _write_ins(experiment_list, best_unit_cell, composition, ins_file):
-            f.write("SFAC C\n")
-            f.write("UNIT 0\n")
+        )
         f.write("HKLF 4\n")  # Intensities
         f.write("END\n")
 
 
 def _parse_compound(composition):
-    """
-    Modified from parse_compund() in xia2/src/xia2/cli/to_shelx.py
-    """
-    elements = [
-        "H",
-        "He",
-        "Li",
-        "Be",
-        "B",
-        "C",
-        "N",
-        "O",
-        "F",
-        "Ne",
-        "Na",
-        "Mg",
-        "Al",
-        "Si",
-        "P",
-        "S",
-        "Cl",
-        "Ar",
-        "K",
-        "Ca",
-        "Sc",
-        "Ti",
-        "V",
-        "Cr",
-        "Mn",
-        "Fe",
-        "Co",
-        "Ni",
-        "Cu",
-        "Zn",
-        "Ga",
-        "Ge",
-        "As",
-        "Se",
-        "Br",
-        "Kr",
-        "Rb",
-        "Sr",
-        "Y",
-        "Zr",
-        "Nb",
-        "Mo",
-        "Tc",
-        "Ru",
-        "Rh",
-        "Pd",
-        "Ag",
-        "Cd",
-        "In",
-        "Sn",
-        "Sb",
-        "Te",
-        "I",
-        "Xe",
-        "Cs",
-        "Ba",
-        "La",
-        "Ce",
-        "Pr",
-        "Nd",
-        "Pm",
-        "Sm",
-        "Eu",
-        "Gd",
-        "Tb",
-        "Dy",
-        "Ho",
-        "Er",
-        "Tm",
-        "Yb",
-        "Lu",
-        "Hf",
-        "Ta",
-        "W",
-        "Re",
-        "Os",
-        "Ir",
-        "Pt",
-        "Au",
-        "Hg",
-        "Tl",
-        "Pb",
-        "Bi",
-        "Po",
-        "At",
-        "Rn",
-        "Fr",
-        "Ra",
-        "Ac",
-        "Th",
-        "Pa",
-        "U",
-        "Np",
-        "Pu",
-    ]
-    result = {}
-    element = ""
-    number = ""
-    composition += "Q"  # Not in any element symbol
-    for c in composition:
-        if c in string.ascii_uppercase:
-            if not element:
-                element += c
-                continue
-            if number == "":
-                count = 1
-            else:
-                count = int(number)
-            if element not in result:
-                if element in elements:
-                    result[element] = 0
-                else:
-                    raise Sorry(f"Element {element} in {composition} not recognised")
-            if element in elements:
-                result[element] += count
-            element = "" + c
-            number = ""
-            if c == "Q":
-                break
-        elif c in string.ascii_lowercase:
-            element += c
-        elif c in string.digits:
-            number += c
+    elements = chemical_elements.proper_caps_list()[:94]
+    m = re.findall(r"([A-Z][a-z]?)(\d*)\s*", composition)
+    if all(e[1] == "" for e in m):
+        # Assume user has just supplied list of elements so UNIT instruction cannot be calculated
+        result = {element: 0 for element, count in m}
+    else:
+        result = {element: int(count or 1) for element, count in m}
```

Comment:
A thought occurs to me.  Does SHELX input support non-stoichiometric formulae?  Should `int` be `float`?

Reply:
It seems it is not an error for `shelxl`, but it would be unconventional. I haven't found any examples of non-integer values after `UNIT` online. Wouldn't this just be represented by the occupancy column for the atom?

As for `shelxt`, it ignores what's in `UNIT` entirely.

Reply:
`UNIT` element count in unit cell not asu. I was multiplying the formula entered by Z. Quite common to have fractions of the formula in the asu but fractions of an atom in the _unit cell_ ?

Reply:
I think we should keep this as int. Convenient to have the composition multiplied by Z :+1:

Reply:
I think I now understand @benjaminhwilliams's comment. Take an example like CCDC 1880252. In that case I think with the current code you'd want to write 'C6 H5 B0.25 F P0.25' if you scaled the data in I-4. So maybe we should have `float` here and convert to int when writing the `UNIT` card? Or allow users to set Z (as in `_cell_formula_units_Z`) to override `space_group().order_z()`?

Reply:
Ah yes, gotcha. Allow float input and convert to int makes sense to me.

Reply:
Thinking some more about this probably the best approach is to *not* allow non-integer values but add a parameter `shelx.zprime` so for the example above the user would enter 'C24 H20 B F4 P' and `zprime=0.25` and get Z = 2. 

In that case should `shelx.composition` should be renamed `shelx.formula`? or should we have both - where `shelx.composition` is just element types and `shelx.formula` is always interpreted as a formula even when there no numbers.